src/exe/model.cc

```diff
@@ -40,6 +41,99 @@
 namespace colmap {
 namespace {
 
+// Aligns the reconstruction to the plane defined by running PCA on the 3D
+// points. The model centroid is at the origin of the new coordinate system
+// and the X axis is the first principal component with the Y axis being the
+// second principal component
+void AlignToPrincipalPlane(Reconstruction& recon, SimilarityTransform3& tform) {
+  // Perform SVD on the 3D points to estimate the ground plane basis
+  auto centroid = recon.ComputeCentroid();
+  Eigen::MatrixXd points(3, recon.NumPoints3D());
+  int pidx = 0;
+  for (const auto point : recon.Points3D()) {
+    points.col(pidx++) = point.second.XYZ() - centroid;
+  }
+  Eigen::Matrix3d basis =
+      points.jacobiSvd(Eigen::ComputeThinU | Eigen::ComputeThinV).matrixU();
+  Eigen::Matrix3d rot_mat;
+  rot_mat << basis.col(0), basis.col(1), basis.col(0).cross(basis.col(1));
+  rot_mat.transposeInPlace();
+
+  tform = SimilarityTransform3(1.0, RotationMatrixToQuaternion(rot_mat),
+                               -rot_mat * centroid);
+
+  // if camera plane ends up below ground then flip basis vectors and create new
+  // transform
+  class Image* test_img = new class Image(recon.Images().begin()->second);
+  tform.TransformPose(&test_img->Qvec(), &test_img->Tvec());
+  if (test_img->ProjectionCenter().z() < 0.0) {
+    rot_mat << basis.col(0), -basis.col(1), basis.col(0).cross(-basis.col(1));
+    rot_mat.transposeInPlace();
+    tform = SimilarityTransform3(1.0, RotationMatrixToQuaternion(rot_mat),
+                                 -rot_mat * centroid);
+  }
+
+  recon.Transform(tform);
+}
+
+// Aligns the reconstruction to the local ENU plane orientation. Rotates the
+// reconstruction such that the x-y plane aligns with the ENU tangent plane at
+// the point cloud centroid and translates the origin to the centroid.
+// If unscaled == true, then the original scale of the model remains unchanged.
+void AlignToENUPlane(Reconstruction& recon, SimilarityTransform3& tform,
```

Comment:
Same here, could be nice to move this function to estimators/coordinate_frame.h and also add a basic unit test.